xrpl/utils/tx_parser/balance_changes.py

```diff
@@ -0,0 +1,231 @@
+                sent the transaction and the affected nodes.
+        """
+        super().__init__(metadata)
+
+        self._parse()
+
+    def _parse(self: Any) -> None:
+        is_valid_metadata(metadata=self.metadata)
+        parsedQuantities = parse_quantities(self.metadata, parse_final_balance)
+
+        result: Dict[str, List[Dict[str, str]]] = {}
+        for k, v in parsedQuantities.items():
+            address = k
+            change = v
+            result[address] = []
+            for obj in change:
+                if isinstance(obj.counterparty, tuple):
+                    obj.counterparty = obj.counterparty[0]
+                result[address].append(
+                    {
+                        "Counterparty": obj.counterparty,
+                        "Currency": obj.currency,
+                        "Value": obj.value,
+                    }
+                )
+
+        self._define_results(result=result)
+
+
+class ParsePreviousBalances(BalanceChanges):
+    """Parse the previous balances of all accounts affected
+    by the transaction before it occurred.
+
+    Usage:
+        Parse all previous balances:
+            ``ParsePreviousBalances(metadata=tx).all_balances``
+        Parse previous balances of the account that sent the transaction:
+            ``ParsePreviousBalances(metadata=tx).source_account_balances``
+        Parse previous balances of the account that received the transaction:
+            ``ParsePreviousBalances(metadata=tx).destination_account_balances``
+                If provided transaction has no field `Destination`:
+                    'Transaction type: `{tx_type}` has no
+                    field 'Destination' or is not included.'
+
+    Attributes:
+        metadata (`Dict[str, Union[str, int, bool, Dict[str, Any]]]`):
+            Transaction metadata including the account that
+            sent the transaction and the affected nodes.
+    """
+
+    def __init__(
+        self: Any, metadata: Dict[str, Union[str, int, bool, Dict[str, Any]]]
+    ) -> None:
+        """
+        Attributes:
+            metadata (`Dict[str, Union[str, int, bool, Dict[str, Any]]]`):
+                Transaction metadata including the account that
+                sent the transaction and the affected nodes.
+        """
+        super().__init__(metadata)
+
+        self._parse()
+
+    def _parse(self: Any) -> None:
+        is_valid_metadata(metadata=self.metadata)
+        balance_changes = ParseBalanceChanges(metadata=self.metadata).all_balances
+        final_balances = ParseFinalBalances(metadata=self.metadata).all_balances
+
+        for account, balances in balance_changes.items():
+            for count, balance in enumerate(balances):
+                final_balances_value = final_balances[account][count]["Value"]
+                final_balances[account][count]["Value"] = str(
+                    Decimal(final_balances_value) - Decimal(balance["Value"])
```

Comment:
Should this be addition? (Are the changes listed where +100 = Add 100 XRP to the previous value?)

Reply:
Example:
`BalanceChanges`:
-100 XRP
`FinalBalance`:
900 XRP
`PreviousBalance`:
900 XRP - -100 XRP = 1000 XRP

If the balance change would increase the balance by 100 the calculation of the `PreviousBalance` would be:
900 XRP - 100 XRP = 800 XRP

So subtraction should be correct :)

Reply:
That makes sense! Thanks for the example :)